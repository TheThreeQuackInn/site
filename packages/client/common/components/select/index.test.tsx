import React from 'react';
import {create} from 'react-test-renderer';
import Select from './';

const options = [
    {
        label: 'Test 1',
        value: 'test-1',
    },
    {
        label: 'Test 2',
        value: 'test-2',
    },
    {
        label: 'Test 3',
        value: 'test-3',
    },
];

describe('Select', () => {
    it('renders correctly', () => {
        const onChange = jest.fn();
        const component = create(<Select label="Testing" value="" onChange={onChange} name="test" options={options} />);
        expect(component).toMatchSnapshot();
    });

    it('renders value correctly', () => {
        const onChange = jest.fn();
        const component = create(
            <Select label="Testing" value="test-2" onChange={onChange} name="test" options={options} />,
        );
        expect(component).toMatchSnapshot();
    });
});
