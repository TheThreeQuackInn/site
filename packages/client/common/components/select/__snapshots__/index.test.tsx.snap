// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<div
  className="w-full"
>
  <label
    className="block uppercase tracking-wide text-gray-700 dark:text-white text-xs font-bold mb-1"
    htmlFor="test"
  >
    Testing
  </label>
  <div
    className="relative"
  >
    <select
      className="block appearance-none w-full bg-gray-200 dark:bg-gray-500 text-gray-700 dark:text-white py-2 px-2 md:px-3 md:py-2 focus:outline-black dark:focus:outline-white rounded leading-tight focus:border-gray-500"
      id="test"
      name="test"
      onChange={[MockFunction]}
      tabIndex={0}
      value=""
    >
      <option
        data-name="Test 1"
        value="test-1"
      >
        Test 1
      </option>
      <option
        data-name="Test 2"
        value="test-2"
      >
        Test 2
      </option>
      <option
        data-name="Test 3"
        value="test-3"
      >
        Test 3
      </option>
    </select>
    <div
      className="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700 dark:text-white"
    >
      <svg
        className="h-5 w-5 inline-block"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Select renders value correctly 1`] = `
<div
  className="w-full"
>
  <label
    className="block uppercase tracking-wide text-gray-700 dark:text-white text-xs font-bold mb-1"
    htmlFor="test"
  >
    Testing
  </label>
  <div
    className="relative"
  >
    <select
      className="block appearance-none w-full bg-gray-200 dark:bg-gray-500 text-gray-700 dark:text-white py-2 px-2 md:px-3 md:py-2 focus:outline-black dark:focus:outline-white rounded leading-tight focus:border-gray-500"
      id="test"
      name="test"
      onChange={[MockFunction]}
      tabIndex={0}
      value="test-2"
    >
      <option
        data-name="Test 1"
        value="test-1"
      >
        Test 1
      </option>
      <option
        data-name="Test 2"
        value="test-2"
      >
        Test 2
      </option>
      <option
        data-name="Test 3"
        value="test-3"
      >
        Test 3
      </option>
    </select>
    <div
      className="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700 dark:text-white"
    >
      <svg
        className="h-5 w-5 inline-block"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
</div>
`;
